components/list/src/provider/mod.rs

```diff
@@ -69,6 +69,8 @@ data_marker!(
     ListFormatterPatterns<'static>,
 );
 
+icu_provider::marker::does_not_deref_to_varule!(ListFormatterPatterns<'_>);
```

Comment:
issue: this is an opt-out, but it should be opt-in

Reply:
Without specialization, I need to implement the trait on everything, and use runtime `Option` to determine if it is actually implement or not.

Reply:
I don't see how specialization is relevant for opt out vs opt in? We're using a macro, the macro can default to generating a blank impl, and one can ask the macro to not do so.

This is my comment above about the reverse default.

Reply:
> I don't see how specialization is relevant for opt out vs opt in

Ideally I would have a blanket impl that returns None and then a small number of specific impls that return Some. Instead, everyone needs to decide how to implement the trait, which I think @robertbastian doesn't like.

Reply:
My objection at least is that people need to explicitly opt out: it's not an objection about the trait architecture, it's an objection about the macro architecture.

Reply:
I did something like this.